scripts/ci-entrypoint.sh

```diff
@@ -91,6 +91,12 @@ wait_for_nodes() {
     done
 
     kubectl wait --for=condition=Ready node --all --timeout=5m
+    if [[ "${EXP_MACHINE_POOL:-}" == "true" ]]; then
+        echo "Waiting for nodes overprovisioned by the MachinePool to be removed"
+        while kubectl get nodes | grep 'NotReady' > /dev/null; do
```

Comment:
how come ` kubectl wait --for=condition=Ready node --all` doesn't cover this?

Reply:
In https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/capz-azure-file-machinepool-1-19/1388742406299455488, I am suspecting that the overprovisioned node was spawned after `kubectl wait --for=condition=Ready node --all` is finished running

Reply:
wouldn't the same thing happen here if that's the case? It would have no NotReady nodes (== all nodes are Ready as checked above) and it would proceed to the next step, making it possible for a NotReady node to appear after

Reply:
Perhaps, we should add a condition to the machine pool to indicate when it reaches a stable state. In #1332, we introduce a `ScaleSetDesiredReplicasCondition` which is true when the desired number of ready nodes is reached. We could add another condition which would describe a stable state with only ready machines which match the desired replica count.